client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -165,8 +165,17 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             sender.setQoS(qos);
             sender.openHandler(senderOpen -> {
                 if (senderOpen.succeeded()) {
-                    LOG.debug("sender open [{}]", sender.getRemoteTarget());
-                    result.complete(senderOpen.result());
+                    LOG.debug("sender open [{}] sendQueueFull [{}]", sender.getRemoteTarget(), sender.sendQueueFull());
+                    // wait on credits a little time, if not already given TODO: configurable waiting time
+                    if(sender.sendQueueFull()) {
+                        ctx.owner().setTimer(10, timerID -> {
+                            LOG.debug("waited 10ms on credits [{}], sendQueueFull [{}]", sender.getCredit(), sender.sendQueueFull());
+                            result.complete();
+                        });
+                    }
+                    else {
```

Comment:
please put the `else` statement on the previous line, adjust your editor/code formatting settings accordingly